README.md

```diff
@@ -959,26 +959,6 @@ _You can enable the following settings in Xcode by running [this script](resourc
 
   </details>
```

Comment:
>  I don't think we should be suggesting that folks create top-level declarations. This feels at-odds with the [Avoid creating non-namespaced global constants and functions](https://github.com/airbnb/swift#namespace-using-enums) rule.

I believe that this rule is synergistic with the [Avoid creating non-namespaced global constants and functions](https://github.com/airbnb/swift#namespace-using-enums) rule that you referenced.

You will notice that this rule explicitly calls out `public` and `internal` in its headline: "Use caseless enums for organizing `public` or `internal` constants and functions into namespaces". I think the "Avoid creating non-namespaced global constants and functions" needs to be read acknowledging that it's specific to `public` and `internal` constants and fucntions.

In the commit to which you linked, the rule at the time also mentions namespacing public constants: "If they are public, put the constant as a static property, so we get nice namespaces." I think this is further evidence that they are synergistic. Eventually that was made it owns rule. I would have to dig deeper into the history to remember it all.

At least one member of the team felt like it was overkill to create a caseless `enum` for private constants. I generally do try to avoid putting functions and properties in the global namespace. But it is a feature of the language. I think allowing `private` functions and properties in the global namespace is a good compromise, since they are isolated to a single file.

Reply:
cc: @jqsilver who wrote the original commit

Reply:
https://github.com/airbnb/swift/pull/106#discussion_r495209375